### DISCUSSION TITLE
Windows OS documentation/reference of the screenshot: command

### DISCUSSION THREAD

**Discussion by henrik-jensen** [ImageMagick/ImageMagick#5106] [Help]
Fragment of a rant from former imagemagick windows user from [Yorkshire](https://www.youtube.com/watch?v=iEIApUNVBKg):
"...You lucky unix hackers with your fancy smancy xwindows `import` command and what not, that seems to have comprehensive documentation. Us poor old windows hackers, trying to make it through the day, only relying on googling random internet post to find fractions of tiny bits of information, in the pursuit of getting only just a small amount of relief to our inhuman torment. Ohhh you lucky bastards, you,- you really have it made, haven't you..." 😏😁

Sooo... anybody know where to find the reference/documentation of the `screenshot:` command used for windows OS?

I can do: `magick screenshot:[100x100+200+300] target.png` to get a cropped screen shot (equivalent to `magick screenshot: -crop 100x100+200+300  target.png`) but I can't control which screen I'm addressing. This results in a cropped `target0.png` and a `target1.png` on a system with two displays. I could do `magick screenshot:[0] -crop 100x100+200+300  target.png` but my [writers cramp prevents me from typing those 6 extra characters](https://www.dailymotion.com/video/x32czn2), so I would like to do something like `magick screenshot:[0][100x100+200+300] target.png`. Tried, it just hangs for a long time.
Besides, my main mission is to find the authorized reference of `screenshot:`.

--Henrik

**Comment by urban-warrior** [comment]
You can find the documentation for the SCREENSHOT format @ https://imagemagick.org/script/formats.php in the Pseudo-image Format section where it succinctly says:

> an image that shows the contents of a computer display.   Under Windows, if you have multiple monitors, it returns a screenshot for each monitor.

As you surmised it enumerates over all the displays and captures the entire screen image.   Given two monitors, you can select just the one with `screenshot:[0]` and subsequently crop it if needed as you have done.  We could add a define to specify the screen, such as `-define screen=1` and there are certainly opportunity to enhance the features of this particular format.

As ImageMagick is open-source, we encourage you or another user to contribute these enhancements in the form of a Git pull request.

**Reply by henrik-jensen** [reply]
Thx.
Yes, succinctly indeed :-)
Anyway `magick screenshot:[0] -crop 100x100+200+300  target.png` is sufficient, honestly.
I was more puzzled about where this mysterious `:[...]` syntax (does it have a name?) and its variations are referenced.
